lib/units/api/controllers/user.js

```diff
@@ -0,0 +1,332 @@
+                  req.user.email
+                , req.user.name
+                , req.user.group
+                )
+              , timeout
+              , wireutil.toDeviceRequirements(requirements)
+              )
+            )
+          ])
+
+          res.status(202).json({
+            success: true
+          , description: 'Device Add request is accepted. Check if device is successfully added using pollingUrl'
+          , pollingUrl: util.format('%s://%s%s/user/devices/%s'
+                                      , req.protocol
+                                      , req.get('host')
+                                      , req.swagger.operation.api.basePath
+                                      , serial
+                                    )
+          })
+
+        } else {
+          res.status(401).json({
+            success: false
+          , description: 'Device is being used or not available'
+          })
+        }
+      } else {
+        res.status(404).json({
+          success: false
+        , description: 'Bad device serial'
+        })
+      }
+    })
+}
+
+function deleteUserDeviceBySerial(req, res) {
+  var serial = req.swagger.params.serial.value
+
+  dbapi.loadDevice(serial)
+    .then(function(device) {
+      if (device) {
+        datautil.normalize(device, req.user)
+        if(device.using && device.owner.email == req.user.email) {
+
+          var requirements = {
+            'serial': {
+              'value': serial,
+              'match': 'exact'
+            }
+          }
+
+          req.options.push.send([
+            device.channel
+          , wireutil.envelope(
+              new wire.UngroupMessage(
+                wireutil.toDeviceRequirements(requirements)
+              )
+            )
+          ])
+
+          res.status(202).json({
+            success: true
+          , description: 'Device Release request is accepted. Check if device is successfully removed using pollingUrl'
+          , pollingUrl: util.format('%s://%s%s/user/devices/%s'
+                                      , req.protocol
+                                      , req.get('host')
+                                      , req.swagger.operation.api.basePath
+                                      , serial
+                                    )
+          })
+
+        } else {
+          res.status(401).json({
+            success: false
+          , description: 'You cannot kick this device'
+          })
+        }
+      } else {
+        res.status(404).json({
+          success: false
+        , description: 'Bad device serial'
+        })
+      }
+    })
+}
+
+function remoteConnectUserDeviceBySerial(req, res) {
+  var serial = req.swagger.params.serial.value
+
+  dbapi.loadDevice(serial)
+    .then(function(device) {
+      if (device) {
+        datautil.normalize(device, req.user)
+
+        if (device.present && device.ready && device.using && device.owner.email === req.user.email) {
```

Comment:
This conditional statement is long. I would like to split to another method (e.g. `device.canConnect(req.user.email)`)